src/core/AudioEngine.cpp

```diff
@@ -808,57 +803,42 @@ void AudioEngine::removePlayHandlesOfTypes(Track * track, PlayHandle::Types type
 
 void AudioEngine::requestChangeInModel()
 {
-	if( s_renderingThread )
-		return;
+	if (s_renderingThread || !m_isProcessing) { return; }
 
-	m_changesMutex.lock();
-	m_changes++;
-	m_changesMutex.unlock();
-
-	m_doChangesMutex.lock();
-	m_waitChangesMutex.lock();
-	if (m_isProcessing && !m_waitingForWrite && !m_changesSignal)
 	{
-		m_changesSignal = true;
-		m_changesRequestCondition.wait( &m_waitChangesMutex );
+		auto lock = std::lock_guard{m_numChangesMutex};
+		++m_numChanges;
 	}
-	m_waitChangesMutex.unlock();
-}
-
-
 
+	auto lock = std::unique_lock{m_runChangesMutex};
+	m_runChangesCond.wait(lock, [this] { return m_runChanges; });
+}
 
 void AudioEngine::doneChangeInModel()
 {
-	if( s_renderingThread )
-		return;
+	if (s_renderingThread || !m_isProcessing) { return; }
```

Comment:
What if `m_isProcessing` changes value between `requestChangeInModel` and `doneChangeInModel`? This would mess up the change counting and notification. The old code used it only to skip waiting, and not to skip counting altogether, which seems more robust to me.

Reply:
If a change is running while ``m_isProcessing`` is set to ``true`` from ``false`` for example, shouldn't the thread that called ``startProcessing`` wait for the change to be complete then?